lua/codeium/util.lua

```diff
@@ -35,8 +37,39 @@ function M.get_newline(bufnr)
 	return enums.line_endings[vim.bo[bufnr].fileformat] or "\n"
 end
 
+-- Get the relative path from the project root
 function M.get_relative_path(bufnr)
-	return vim.fn.fnamemodify(vim.api.nvim_buf_get_name(bufnr), ":")
+	local buf_path = vim.api.nvim_buf_get_name(bufnr)
+	local start_path = M.get_project_root()
+	return Path:new(buf_path):make_relative(start_path)
+end
+
+local cached_roots = {}
+function M.get_project_root()
+	local cwd = vim.fn.getcwd()
+
+	if cached_roots[cwd] then
+		return cached_roots[cwd]
+	end
+
+	-- From the CWD, walk up the tree looking for a directory that contains one of the project root files
+	local candidates = config.options.project_root_paths
```

Comment:
Maybe the source of candidates should be `vim.lsp.buf.list_workspace_folders()`?

Reply:
That seems like a good idea. Personally I'm actually still on coc.nvim so I'll probably keep the project_root_paths option as a fallback when `list_workspace_folders` returns nothing.

Reply:
Sounds reasonable.